src/components/TokenBalanceInput/index.tsx

```diff
@@ -0,0 +1,62 @@
+import { FC } from "react"
+import {
+  Box,
+  Button,
+  Icon,
+  InputGroup,
+  InputLeftElement,
+  InputProps,
+  InputRightElement,
+} from "@chakra-ui/react"
+import NumberInput, { NumberInputValues } from "../NumberInput"
+import { Body3 } from "../Typography"
+
+export interface TokenBalanceInputProps extends InputProps {
+  icon: any
```

Comment:
I think we should avoid using 'any' type as much as possible, because it is [similar to putting an @ts-ignore around every single use of the variable](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#any). Because of that by the quick look I'm not able to tell if we should use string here or Icon component or something else.

I think we are using the Icons that we created from `createElement` function from Chakra (correct me if I'm wrong) so we can define type like this:
```suggestion
  icon: ReturnType<typeof createIcon>
```